backend/accounts/views.py

```diff
@@ -38,38 +43,35 @@ def create(self, request, *args, **kwargs):
 
 
 @extend_schema(tags=['Accounts'])
-class UserUpdateView(generics.RetrieveUpdateDestroyAPIView):
+class UserUpdateView(viewsets.ModelViewSet):
     """
     View for get or update the user personal information and user delete \n
         - API endpoint: /api/accounts/user/<int:pk>/
     """
 
     queryset = User.objects.all()
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticatedOrReadOnly, IsOwnerOfObject]
     serializer_class = UpdateUserSerializer
     parser_classes = [JSONParser, MultiPartParser, FormParser]
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = UserFilter
+    http_method_names = ['get', 'put', 'patch', 'delete']
 
-    def retrieve(self, request, *args, **kwargs):
-        """
-        View for update the user personal information and user delete \n
-            - API endpoint for get user information. : /api/accounts/user/<int:pk>/
-        """
-        user = self.get_object()
+    def list(self, request, *args, **kwargs):
```

Comment:
The `list` endpoint is meant to be a "get all entities" endpoint. The standard API practice is something like this:
- GET `/api/accounts/user` -> gets all users (public route, can optionally make use of query params)
- GET `/api/accounts/user/<id>` -> gets details about a specific user (public route)
- GET `/api/accounts/user/<id>/private` -> gets all sensitive details about a specific user (private endpoint - does not have to be named private in the URL, though, could be anything else, especially if it's related to some functionality)
- POST `/api/accounts/user`
- PUT, PATCH, DELETE `api/accounts/user` -> private endpoint

Reply:
So therefore, I would recommend you to restructure it like this unless you have some other suggestion, or if I missed anything?

Reply:
Point taken. The code was refactored as follows:
 - The "list" endpoint now is public and shows user information requested by frontend.
    If there is no username as a query param the result is as bellow:
      ```
      [
        {
          "username": "simo1",
          "avatar": null
        },
        {
          "username": "simo",
          "avatar": "http://127.0.0.1:8000/media/avatars/user_simo/assassins.jpg"
        }
      ]
      ```
    If username present as a query param:
    ```
    {
      "username": "simo",
      "avatar": "http://127.0.0.1:8000/media/avatars/user_simo/assassins.jpg"
    }

 - The retrieve endpoint shows full information for registered user:
    ```
    {
      "id": 17,
      "username": "simo",
      "email": "simo@simo.com",
      "first_name": "Simo",
      "last_name": "Simov",
      "phone_number": null,
      "city": null,
      "shipping_address": "",
      "avatar": "http://127.0.0.1:8000/media/avatars/user_simo/assassins.jpg"
    }
   ```

I hope I understood the comments correctly and the changes in the code are in the right direction.

Reply:
@RyotaMitaraiWeb, are the changes sufficient? Is what you requested satisfied?

Reply:
Based on tinkering with it in Swagger, it looks perfect, great job! I will move forward with implementing it on the front end